### PR TITLE
feat!: include dual cjs/esm in package exports

### PR DESCRIPTION
**Pull request by userquin**
### Description

This PR include:
- add postbuild script: delete `d.mts` files, rename `.mjs` to `.js` and patch `.cjs` files
- split package exports in require/import
- include publint

### Additional context

Current types (check https://arethetypeswrong.github.io/?p=vite-plugin-inspect%400.7.42 and https://publint.dev/vite-plugin-inspect@0.7.42):

imagen

With this PR:

imagen

imagen

### CODE REVIEW

**Comment by userquin** [comment]
I think we should revert las commit...

File: package.json [hunk 1]
```diff
@@ -20,19 +20,29 @@
   ],
   "exports": {
     ".": {
-      "types": "./dist/index.d.ts",
-      "import": "./dist/index.mjs",
-      "require": "./dist/index.cjs"
+      "import": {
```
[the remark this answers is not in the document]
Comment (userquin) [reply package.json:23]: thx Kevin

[the remark this answers is not in the document]
Comment (userquin) [reply package.json:23]: how about last commit?

[the remark this answers is not in the document]
Comment (sxzz) [reply package.json:23]: We can revert the last commit https://github.com/antfu/vite-plugin-inspect/pull/99/commits/81a59e89bde5102cd1d0ff16639304f9e0c6f520

**Review by sxzz — APPROVED** [review]

**Review by antfu — APPROVED** [review]

File: scripts/postbuild.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+rmSync(resolve('./dist/nuxt.d.mts'))
+
+readdirSync('./dist/shared')
+  .filter(file => file.endsWith('.d.mts'))
+  .forEach((file) => {
+    rmSync(resolve('./dist/shared', file))
+  })
+
+renameSync(resolve('./dist/index.mjs'), resolve('dist/index.js'))
+renameSync(resolve('./dist/nuxt.mjs'), resolve('dist/nuxt.js'))
```
Comment (antfu) [inline scripts/postbuild.ts:23]: Why rename mjs to js? I'd actually prefer mjs to be explicit when possible

Comment (userquin) [reply scripts/postbuild.ts:23]: We need to add the types (type is module):

imagen

Including types:

imagen

**Pull request merged**